src/components/ProductCard/index.jsx

```diff
@@ -1,8 +1,169 @@
-import React from "react";
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import { useWishlist } from "../../Contexts/WishlistContext";
+import utils from "../../utils";
 import Button from "../Header/Button";
+import { toast } from "react-toastify";
 import "./product-card.css";
-// imagePath={imageUrl} badgeText={'New Arrival'} title = {'One ear 100 headphones black'} price={1890} actualPrice={2099} discountPercentage={10} 
-const ProductCard = ({variant,product,cartActionBtnContainer}) => {
+import { useAuth } from "../../Contexts/AuthDialogContext";
+import { useModal } from "../../Contexts/ModalContext";
+import { useLocation } from "react-router-dom";
+import { useCart } from "../../Contexts/CartContext";
```

Comment:
Suggestion: you can use absolute import , way of importing file looks more clean and understandable .
reference: you can read about it here https://javascript.plainenglish.io/why-and-how-to-use-absolute-imports-in-react-d5b52f24d53c